scripts/partition/plot.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Plot results of comparing streaming dataset partitioning algorithms."""
+
+from argparse import ArgumentParser, Namespace
+from typing import Dict, List, Tuple
+
+from matplotlib import pyplot as plt
+
+mark_sizes = [
+    ('CIFAR-10 (50K)', 50_000, '#dfd'),
+    ('ImageNet (1.2M)', 1_281_167, '#cfc'),
+    ('C4 (365M)', 364_868_892, '#8c8'),
+    ('LAION-5B (5.85B)', 5_850_000_000, '#484'),
+]
+
+mark_times = [
+    ('Instant (1s)', 1, '#bdf'),
+    ('Fast (10s)', 10, '#9ce'),
+    ('Slow (60s)', 60, '#08f'),
+]
+
+algo_colors = {
+    'orig': 'red',
+}
+
+
+def parse_args() -> Namespace:
+    """Parse command-line arguments.
+
+    Returns:
+        Namespace: Command-line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument('--in', type=str, required=True, help='Partition benchmarking results')
```

Comment:
Who(or which script) writes the benchmarking results ?

Reply:
also just curious about the PR overall -- this is to let users benchmark how long different partitions/shuffles take with different configurations of NCN, physical nodes, ranks, workers, batch size, etc. right? as well as see how this performance varies across some typical datasets?

Reply:
For both partitioning and shuffling, there is a `bench.py` whose benchmarking results are piped to a file which is then plotted by `plot.py`

Reply:
Right, given various factors you've listed, see what the scalability curve for generating the epoch looks like on your specific system, exactly, without having to actually generate many fake datasets of all that range of sizes and train on them.